### PR TITLE
Atom syndication, pages, and archives for indexes and tag indexes

### PR DESCRIPTION
**Pull request by Aeyoun**
https://tools.ietf.org/html/rfc4287

plus extensions:
https://tools.ietf.org/html/rfc5005#section-3 (pagination)
https://tools.ietf.org/html/rfc5005#section-4 (archives)

### CODE REVIEW

File: nikola/conf.py.in [hunk 1]
```diff
@@ -557,7 +557,7 @@ INDEX_READ_MORE_LINK = ${INDEX_READ_MORE_LINK}
 RSS_READ_MORE_LINK = ${RSS_READ_MORE_LINK}
 
 # Append a URL query to the RSS_READ_MORE_LINK and the //rss/item/link in
-# RSS feeds. Minimum example for Piwik "pk_campaign=rss" and Google Analytics
+# Atom and feeds. Minimum example for Piwik "pk_campaign=rss" and Google Analytics
```
Comment (Kwpolska) [inline nikola/conf.py.in]: make it clear: Atom and *RSS* feeds.

File: nikola/data/themes/base/assets/xml/atom.xsl [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+<head>
+<meta charset="UTF-8"/>
+<meta name="viewport" content="width=device-width"/>
+<title><xsl:value-of select="feed/title"/> (Atom feed)</title>
+<style><![CDATA[html{margin:0;pdding:0;}body{color:hsl(180,1%,31%);font-family:Helvetica,Arial,sans-serif;font-size:17px;line-height:1.4;margin:5%;max-width:35rem;padding:0;}input{min-width:20rem;margin-left:.2rem;padding-left:.2rem;padding-right:.2rem;}ol{list-style-type:disc;padding-left:1rem;}h2{font-size:22px;font-weight:inherit;}]]></style>
+</head>
+<body>
+<h1><xsl:value-of select="feed/title"/> (Atom feed)</h1>
+<p>This is an Atom feed. To subscribe to it, copy its address and paste it when your feed reader asks for it. It will be updated periodically in your reader. New to feeds? <a href="https://duckduckgo.com/?q=how+to+get+started+with+rss+feeds" title="Search on the web to learn more">Learn more</a>.</p>
+<p>
```
Comment (Kwpolska) [inline nikola/data/themes/base/assets/xml/atom.xsl]: The search query still says `rss`.

On a side note, could you make it possible to discourage people from using feeds for subpages (index-1 etc.)? I have absolutely no idea how this all works and if this is possible; maybe you could create two XSL files with different text for the two cases.

Comment (Aeyoun) [inline nikola/data/themes/base/assets/xml/atom.xsl]: Intentional. The results were better than when changing the query to Atom. You’d be hard-pressed to find something calling itself an RSS reader that didn’t support Atom anyways.

That is the job of `<link rel="current">` inside the non-current feeds. Could display the `current address` itself. Would be a bit more complicated as current doesn’t contain current only self.

File: nikola/conf.py.in [hunk 3]
```diff
@@ -753,12 +757,21 @@ COMMENT_SYSTEM_ID = ${COMMENT_SYSTEM_ID}
 # links to it.  Set this to False to disable everything RSS-related.
 # GENERATE_RSS = True
 
+# By default, Nikola does not generates Atom files for indexes and links to
+# them. Generate Atom for tags by setting TAG_PAGES_ARE_INDEXES to True. 
+# Atom feeds are built based on INDEX_DISPLAY_POST_COUNT and not FEED_LENGHT
```
Comment (Kwpolska) [inline nikola/conf.py.in]: LENGTH*

File: nikola/conf.py.in [hunk 4]
```diff
@@ -753,12 +757,21 @@ COMMENT_SYSTEM_ID = ${COMMENT_SYSTEM_ID}
...
+# Switch between plain-text summaries and full HTML content using the
+# RSS_TEASER option. RSS_LINKS_APPEND_QUERY is also respected. Atom feeds
+# are generated even for old indexes and have pagination link relations
+# between each other. Old Atom feeds were no changes are marked as archived.
```
Comment (Kwpolska) [inline nikola/conf.py.in]: **with** no changes*?

File: nikola/data/themes/base/assets/xml/atom.xsl [hunk 5]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:dc="http://purl.org/dc/elements/1.1/" version="1.0">	
+<xsl:output method="xml"/>
+<xsl:template match="/">
+<html xmlns="http://www.w3.org/1999/xhtml" lang="en">
+<head>
+<meta charset="UTF-8"/>
+<meta name="viewport" content="width=device-width"/>
+<title><xsl:value-of select="feed/title"/> (Atom feed)</title>
+<style><![CDATA[html{margin:0;pdding:0;}body{color:hsl(180,1%,31%);font-family:Helvetica,Arial,sans-serif;font-size:17px;line-height:1.4;margin:5%;max-width:35rem;padding:0;}input{min-width:20rem;margin-left:.2rem;padding-left:.2rem;padding-right:.2rem;}ol{list-style-type:disc;padding-left:1rem;}h2{font-size:22px;font-weight:inherit;}]]></style>
```
Comment (Kwpolska) [inline nikola/data/themes/base/assets/xml/atom.xsl]: padding*

(also broken in rss.xsl)

**Comment by Kwpolska** [comment]
Couldn’t we handle the tags without forcing tag indexes?

**Pull request merged**